### PR TITLE
Allows you to change the blade color of toy swords with a screwdriver

### PR DESCRIPTION
**Pull request by ShizCalev**
Requires #70364

:cl: ShizCalev
qol: You can now change the color of toy energy swords with a screwdriver!
/:cl:

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -415,6 +415,26 @@
 	playsound(user ? user : src, active ? 'sound/weapons/saberon.ogg' : 'sound/weapons/saberoff.ogg', 20, TRUE)
 	return COMPONENT_NO_DEFAULT_MESSAGE
 
+/obj/item/toy/sword/examine(mob/user)
+	. = ..()
+	. += span_notice("It has a lighting dial currently set to [saber_color] which looks like it can be turned with a <b>screwdriver</b>.")
+
+/obj/item/toy/sword/screwdriver_act(mob/living/user, obj/item/tool)
+	. = TOOLACT_SUCCESSFUL
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:423]: Just `return TOOL_ACT_SUCCESSFUL` in the two branches

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -415,6 +415,26 @@
...
+			saber_color = "blue"
+		if("blue")
+			saber_color = "green"
+		if("green")
+			saber_color = "purple"
+		if("purple")
+			saber_color = "red"
+		else
+			return
+	to_chat(user, span_notice("You turn [src]'s lighting dial to [saber_color]."))
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:435]:
```suggestion
	balloon_alert(user, "changed to [saber_color]")
```

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -415,6 +415,26 @@
 	playsound(user ? user : src, active ? 'sound/weapons/saberon.ogg' : 'sound/weapons/saberoff.ogg', 20, TRUE)
 	return COMPONENT_NO_DEFAULT_MESSAGE
 
+/obj/item/toy/sword/examine(mob/user)
+	. = ..()
+	. += span_notice("It has a lighting dial currently set to [saber_color] which looks like it can be turned with a <b>screwdriver</b>.")
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:420]: Add a contextual screentip too: [`/datum/element/contextual_screentip_tools`](https://github.com/tgstation/tgstation/blob/master/code/datums/elements/screentips/README.md#datumelementcontextual_screentip_tools)

**Review by Mothblocks — DISMISSED** [review]
Neat

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -415,6 +415,26 @@
...
+		if("red")
+			saber_color = "blue"
+		if("blue")
+			saber_color = "green"
+		if("green")
+			saber_color = "purple"
+		if("purple")
+			saber_color = "red"
+		else
+			return
```
Comment (ShizCalev) [inline code/game/objects/items/toys.dm:434]:
```suggestion
			return TOOL_ACT_SUCCESSFUL
```

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (ShizCalev) [reply code/game/objects/items/toys.dm:423]: trailing returns suck dick.

File: code/game/objects/items/toys.dm (hunk 3 above)
Comment (ShizCalev) [reply code/game/objects/items/toys.dm:420]: i dig it. will do when i'm not drunk.

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (Mothblocks) [reply code/game/objects/items/toys.dm:422]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STANDARDS.md#dot-variable-

Comment (ShizCalev) [reply code/game/objects/items/toys.dm:422]: fair enough! at the end of the day same thing, so i'll change it over tomorrow :)

**Comment by TheGreatValvatorez** [comment]
Can't you just make this a multitool thing?
Instead of always turning it into rainbow it pops up a radial wheel allowing you to pick your color

**Comment by ShizCalev** [comment]
> Can't you just make this a multitool thing? Instead of always turning it into rainbow it pops up a radial wheel allowing you to pick your color

The multitool thing is a bit of a hidden feature since you can do it with normal eswords as well.

**Pull request merged**